### PR TITLE
build: prettier, eslint, and package updates

### PR DESCRIPTION
**Pull request by LeoDog896**
- Finally applies https://github.com/parcel-bundler/parcel/pull/8868 to deduplicate assets
- Integrates prettier and eslint ([dprint](https://dprint.dev/) is also a viable in-rust alternative for prettier)

### CODE REVIEW

**Comment by huacnlee** [comment]
LTGM

**Review by huacnlee — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on: [push, pull_request]
```
Comment (Jamalam360) [inline .github/workflows/publish.yml:4]: What's causing _some_ yaml keys to be quoted and some to not be? Is there a way to configure prettier to only put quotes when required?

**Review by Jamalam360 — COMMENTED** [review]
Looks good to me, just a nitpick

File: .github/workflows/publish.yml (hunk 1 above)
Comment (LeoDog896) [reply .github/workflows/publish.yml:4]: Actually, good question! I tinkered with Prettier a bit and it only quotes things that previously had quotes - we can drop the ones that only have 1 word w/out issue

**Review by Jamalam360 — APPROVED** [review]

**Comment by tomtau** [comment]
@LeoDog896 seems the publishing pipeline fails now: https://github.com/pest-parser/site/actions/runs/6109107940/job/16579443190

**Pull request merged**